modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/transformers/ActionsComponentPropsTransformer.js

```diff
@@ -17,6 +17,44 @@ import {render} from '@liferay/frontend-js-react-web';
 import SidebarPanel from '../components/SidebarPanel';
 import SidebarPanelInfoView from '../components/SidebarPanelInfoView/SidebarPanelInfoView';
 import SidebarPanelMetricsView from '../components/SidebarPanelMetricsView';
+import {OPEN_PANEL_VALUE} from '../utils/constants';
+
+const handlePanelStateFromSession = async () => {
+	const _panelState = await Liferay.Util.Session.get(
+		'com.liferay.content.dashboard.web_panelState'
+	);
+
+	if (_panelState !== OPEN_PANEL_VALUE) {
+		return;
+	}
+
+	const {fetchURL, portletNamespace, rowId} = await Liferay.Util.Session.get(
+		'com.liferay.content.dashboard.web_panelCurrentItemInfo'
+	);
+
+	showSidebar({
+		View: SidebarPanelInfoView,
+		fetchURL,
+		portletNamespace,
+	});
+
+	selectRow(portletNamespace, rowId);
+};
+
+const handleSessionOnSidebarOpen = ({fetchURL, portletNamespace, rowId}) => {
+	Liferay.Util.Session.set(
```

Comment:
Could we change the panel state value only if it has changed? We could save some petitions.

Reply:
Yeah, I like this. However we are re-fetching the item in all cases so this will a minor improvement, we could try in the near future to complete the optimization.